src/server/pps/server/api_server_integration_test.go

```diff
@@ -137,3 +141,95 @@ func TestRerunPipeline(t *testing.T) {
 	require.Equal(t, int64(0), jobInfo.DataRecovered)
 	require.Equal(t, int64(0), jobInfo.DataFailed)
 }
+
+// TestSidecarMetrics tests that sidecar metrics are available.
+func TestSidecarMetrics(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	t.Parallel()
+	c, namespace := minikubetestenv.AcquireCluster(t)
+
+	inputRepo := "input"
+	require.NoError(t, c.CreateRepo(pfs.DefaultProjectName, inputRepo))
+
+	var pipelineTemplate1 = `{
+		"pipeline": {
+			"project": {
+				"name": "{{.ProjectName | js}}"
+			},
+			"name": "{{.PipelineName | js}}"
+		},
+		"transform": {
+			"cmd": ["cp", "-r", "/pfs/in/foo", "/pfs/out"]
+		},
+		"input": {
+			"pfs": {
+				"project": "default",
+				"repo": "{{.RepoName | js}}",
+				"glob": "/*",
+				"name": "in"
+			}
+		},
+		"autoscaling": false
+	}`
+	tmpl1, err := template.New("pipeline").Parse(pipelineTemplate1)
+	require.NoError(t, err, "template must parse")
+	var buf bytes.Buffer
+	require.NoError(t, tmpl1.Execute(&buf, struct {
+		ProjectName, PipelineName, RepoName string
+	}{pfs.DefaultProjectName, "pipeline1", inputRepo}), "template must execute")
+	_, err = c.PpsAPIClient.CreatePipelineV2(ctx, &pps.CreatePipelineV2Request{
+		CreatePipelineRequestJson: buf.String(),
+	})
+	require.NoError(t, err, "creating pipeline1 must succeed")
+
+	// pipeline2 pulls metrics from pipeline1
+	var pipelineTemplate2 = `{
+		"pipeline": {
+			"project": {
+				"name": "{{.ProjectName | js}}"
+			},
+			"name": "{{.PipelineName | js}}"
+		},
+		"transform": {
```

Comment:
I think this is really a spout.

Reply:
Kind of, expect that I don’t want to run it until after `pipeline1` has at least started up and started publishing metrics. Also, a spout container needs to use the API to interact with Pachyderm rather than just writing to `/pfs/out/`, and means a more complex image.

You’re not wrong, but I think that this is simpler.